apps/sms/views/conversation/js/conversation.js

```diff
@@ -1159,7 +1165,8 @@ var ConversationView = {
       // If we're leaving a thread's message view,
       // ensure that the thread object's unreadCount
       // value is current (set = 0)
-      if (Threads.active) {
+      if (this.currentThread) {
+        // TODO: Will need to replace with Conversation service method.
```

Comment:
So maybe we can just set thread as read while entering the panel and unify with https://github.com/steveck-chung/gaia/blob/message-remove-threads-dependency/apps/sms/views/conversation/js/conversation.js#L604. It will become a single entry that calling a conversation service method and change the thread property, then fire a thread change event to inbox view for UI update, wdyt?